packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart

```diff
@@ -252,6 +252,34 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
     return paths.map((dynamic path) => XFile(path as String)).toList();
   }
 
+  @override
+  Future<List<XFile>> getMedia({
+    MediaOptions? options,
+  }) async {
+    options ??= MediaOptions(allowMultiple: true);
+    final ImageOptions imageOptions = options.imageOptions;
+    final int? imageQuality = imageOptions.imageQuality;
+    final double? maxImageWidth = imageOptions.maxWidth;
+    final double? maxImageHeight = imageOptions.maxHeight;
+
+    final Map<String, dynamic> args = <String, dynamic>{
+      'maxImageWidth': maxImageWidth,
+      'maxImageHeight': maxImageHeight,
+      'imageQuality': imageQuality,
+      'allowMultiple': options.allowMultiple,
+    };
+    List<String>? paths;
+    paths = await _channel
+        .invokeMethod<List<dynamic>?>(
+          'pickMedia',
+          args,
+        )
+        .then((List<dynamic>? paths) =>
+            paths?.map((dynamic path) => path as String).toList());
+
+    return paths?.map((String path) => XFile(path)).toList() ?? <XFile>[];
```

Comment:
Why the double `map`, rather than just having the above map do `=> XFile(path as String)`?